examples/image.md

```diff
@@ -0,0 +1,236 @@
+}
+```
+
+The output would be similar to: 
+
+<p>
+<img src = "assets/multiple_image.png" width = "800" height = "400"/>
+</p>
+
+### 3. Multiple Image Arma Mat
+
+Since we would be logging mlpack's metrics there are chances when we would have to visualize images stored in `arma::mat` and thus you can use the following API to directly log Armadillo matrices:
+
+```cpp
+void SummaryWriter<Filewriter>::Image(const std::string& tag,
+                int step,
+                arma::Mat<eT>& matrix,
+                mlpack::data::ImageInfo& info,
+                Filewriter& fw,
+                const std::string& displayName,
+                const std::string& description)
+```
+
+The API accepts `tag`, `step`, `encodedImage` (An `arma::mat` which has image data) and then the `mlpack::data::ImageInfo` object along with filewriter object and Name and description.
+
+Following is a snippet that would log a multiple image which is stored in `arma::mat` for 1 step in temp directory.
+
+```cpp
+#include <mlboard/mlboard.hpp>
+#include <iostream>
+#include <chrono> 
+#include <ctime> 
+#include <future>
+
+int main()
+{
+    std::chrono::time_point<std::chrono::system_clock> start, end; 
+    start = std::chrono::system_clock::now(); 
+    mlboard::FileWriter f1("temp");
+    
+    // Following line are just there to come up with a arma::mat of images.
+    // If you aldeardy have a matrix with images you can avoid the following lines.
+    arma::Mat<unsigned char> matrix;
+    mlpack::data::ImageInfo info;
+    std::vector<std::string> files = {"./assets/single_image.png",
+         "./assets/mltiple_image.png"};
+
+    // Creating the matrix which has image.
+    mlpack::data::Load(files, matrix, info, false);
+
+    // Now we can log the matrix.
+    mlboard::SummaryWriter<mlboard::FileWriter>::Image(
+         "Multiple Image", 1, matrix, info, f1, "Sample Multiple Image",
+         "This is a Sample image logged using mlboard.");
+    f1.Close();
+    end = std::chrono::system_clock::now();
+    std::chrono::duration<double> elapsed_seconds = end - start;
+    std::time_t end_time = std::chrono::system_clock::to_time_t(end); 
+    
+    std::cout << "finished computation at " << std::ctime(&end_time) 
+              << "elapsed time: " << elapsed_seconds.count() << "s\n"; 
+}
+```
+
+The output would be similar to [Log multiple image](#2-multiple-image)
+
+Note : This api takes quite a time to log the image since it first have to process the image stored in `arma::mat`. The way it process is, it saves the image using `mlpack::data::Load()` api for image and then reads it back in encoded format. It is something similar to what mxnet logging tool does [here](https://github.com/TeamHG-Memex/tensorboard_logger/blob/0ec6f1147b7cea8c4b68d26637781ccd8ec036b9/tensorboard_logger/tensorboard_logger.py#L157). Hence it is recomended to log image summary using a `arma::mat` in a seperate thread to avoid increase in training time.
```

Comment:
https://stackoverflow.com/questions/646286/how-to-write-png-image-to-string-with-the-pil

Reply:
> They used a buffer to save the images it's in memory. 

Yes, thus the statement -> `It is something similar` , but not exactly because we can't do that currently or may be I can't find out a way of doing like that using mlpack image support.